jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererCodeCoverageTrackerTest.kt

```diff
@@ -56,54 +57,49 @@ import software.aws.toolkits.jetbrains.services.codewhisperer.util.CodeWhisperer
 import software.aws.toolkits.jetbrains.services.telemetry.NoOpPublisher
 import software.aws.toolkits.jetbrains.services.telemetry.TelemetryService
 import software.aws.toolkits.jetbrains.settings.AwsSettings
+import software.aws.toolkits.jetbrains.utils.rules.JavaCodeInsightTestFixtureRule
 import software.aws.toolkits.jetbrains.utils.rules.PythonCodeInsightTestFixtureRule
 import software.aws.toolkits.telemetry.CodewhispererCompletionType
 import software.aws.toolkits.telemetry.CodewhispererLanguage
 
-class CodeWhispererCodeCoverageTrackerTest {
-    private class TestCodePercentageTracker(
+internal open class CodeWhispererCodeCoverageTrackerTestBase {
+    protected class TestCodePercentageTracker(
         timeWindowInSec: Long,
         language: CodewhispererLanguage,
         rangeMarkers: MutableList<RangeMarker> = mutableListOf(),
         codeCoverageTokens: MutableMap<Document, CodeCoverageTokens> = mutableMapOf()
     ) : CodeWhispererCodeCoverageTracker(timeWindowInSec, language, rangeMarkers, codeCoverageTokens)
 
-    private class TestTelemetryService(
+    protected class TestTelemetryService(
         publisher: TelemetryPublisher = NoOpPublisher(),
         batcher: TelemetryBatcher
     ) : TelemetryService(publisher, batcher)
 
     @Rule
     @JvmField
-    var projectRule = PythonCodeInsightTestFixtureRule()
+    val pythonProjectRule = PythonCodeInsightTestFixtureRule()
 
     @Rule
     @JvmField
-    val mockClientManagerRule = MockClientManagerRule()
+    val javaProjectRule = JavaCodeInsightTestFixtureRule()
```

Comment:
same comment as last time i reviewed this:

why not just make the test base abstract and the project rule abstract?

Reply:
sounds good, let me update it